lib/dsl/lexer.js

```diff
@@ -23,8 +23,51 @@ const Lexer = chevrotain.Lexer;
 
 const tokens = {};
 
+const baseNamePattern = 'baseName';
 const namePattern = /[a-zA-Z_][a-zA-Z_\-\d]*/;
-// All the names tokens of the pegjs implementation have been merged into
+// using start anchor because we are matching with this regexp directly in a custom pattern.
+const namePatternWithStartDigit = /^[\da-zA-Z_][a-zA-Z_\-\d]*/;
```

Comment:
Some rules should be enforced here, whether to use `\d` or `0-9`. Having both is not that pretty. By the way, why isn't it ` const namePatternWithStartDigit = /^[\da-zA-Z_][a-zA-Z_\-\d]*$/; ` the `$` sign marking the end of the local matching)

Reply:
I agree.